File: app/src/main/java/code/fortomorrow/riceapp/ConfirmFinalOrderActivity.java
package code.fortomorrow.riceapp;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.database.ValueEventListener;

import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.HashMap;

import code.fortomorrow.riceapp.Model.Users;
import code.fortomorrow.riceapp.Prevalent.Prevalent;

public class ConfirmFinalOrderActivity extends AppCompatActivity {
    private EditText nameEditText, phoneEditText, addressEditText, cityEditText;
    private Button confirmOrderBtn;
    private String totalAmount = "";



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_confirm_final_order);

        totalAmount = getIntent().getStringExtra("Total price");

        Toast.makeText(this,"Total Ammount :  "+totalAmount+"$",Toast.LENGTH_LONG).show();
        confirmOrderBtn = findViewById(R.id.confirm_final_order_btn);
        nameEditText = findViewById(R.id.shippment_name);
        phoneEditText = findViewById(R.id.shippment_phone_number);
        addressEditText = findViewById(R.id.shippment_address);
        cityEditText = findViewById(R.id.shipment_city);
        final DatabaseReference databaseReference = FirebaseDatabase.getInstance().getReference().child("Users").child(Prevalent.currentOnlineUser.getPhone());
        databaseReference.addValueEventListener(new ValueEventListener() {
            @Override
            public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
                String name = dataSnapshot.getValue(Users.class).getName();
                nameEditText.setText(name);
            }

            @Override
            public void onCancelled(@NonNull DatabaseError databaseError) {

            }
        });

        confirmOrderBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                check();
            }
        });
    }



    private void check() {
        if(TextUtils.isEmpty(nameEditText.getText().toString())){
            nameEditText.setError("Please provide your full name");
        }
        else if(TextUtils.isEmpty(phoneEditText.getText().toString())){
            phoneEditText.setError("Please provide your phone number");
        }
        else if(TextUtils.isEmpty(addressEditText.getText().toString())){
            addressEditText.setError("Please provide your phone number");
        }
        else if(TextUtils.isEmpty(cityEditText.getText().toString())){
            cityEditText.setError("Please provide your phone number");
        }
        else {
            confirmOrder();
        }
    }

    private void confirmOrder() {
        final String saveCurrentDate, saveCurrentTime;
        Calendar callForDate = Calendar.getInstance();
        SimpleDateFormat currentDate = new SimpleDateFormat("MMM dd, yyyy");
        saveCurrentDate = currentDate.format(callForDate.getTime());
        SimpleDateFormat currentTime = new SimpleDateFormat("HH:mm:ss a");
        saveCurrentTime = currentDate.format(callForDate.getTime());

        final DatabaseReference ordersRef = FirebaseDatabase.getInstance().getReference()
                .child("Orders")
                .child(Prevalent.currentOnlineUser.getPhone());
        HashMap<String,Object> ordersMap = new HashMap<>();
        ordersMap.put("totalAmount",totalAmount);
        ordersMap.put("name",nameEditText.getText().toString());
        ordersMap.put("phone",phoneEditText.getText().toString().toString());
        ordersMap.put("adress",addressEditText.getText().toString());
        ordersMap.put("city",cityEditText.getText().toString());
        ordersMap.put("date",saveCurrentDate);
        ordersMap.put("time",saveCurrentTime);
        ordersMap.put("state","not shipped");


        ordersRef.updateChildren(ordersMap).addOnCompleteListener(new OnCompleteListener<Void>() {
            @Override
            public void onComplete(@NonNull Task<Void> task) {
                if(task.isSuccessful()){
                    FirebaseDatabase.getInstance().getReference()
                            .child("Cart List")
                            .child("User View")
                            .child(Prevalent.currentOnlineUser.getPhone())
                            .removeValue()
                            .addOnCompleteListener(new OnCompleteListener<Void>() {
                                @Override
                                public void onComplete(@NonNull Task<Void> task) {
                                    if(task.isSuccessful()){
                                        Toast.makeText(ConfirmFinalOrderActivity.this,"Your final Order has been placed successfully",Toast.LENGTH_LONG).show();
                                        Intent intent = new Intent(ConfirmFinalOrderActivity.this,HomeActivity.class);
                                        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK|Intent.FLAG_ACTIVITY_CLEAR_TASK);
                                        finish();
                                    }
                                }
                            });
                }
            }
        });
    }
}

File: app/src/main/java/code/fortomorrow/happyshopping/ConfirmFinalOrderActivity.kt
package code.fortomorrow.happyshopping

import android.content.Intent
import android.os.Bundle
import android.text.TextUtils
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import code.fortomorrow.happyshopping.model.Users
import code.fortomorrow.happyshopping.prevalent.Prevalent
import com.google.firebase.database.DataSnapshot
import com.google.firebase.database.DatabaseError
import com.google.firebase.database.FirebaseDatabase
import com.google.firebase.database.ValueEventListener
import java.text.SimpleDateFormat
import java.util.Calendar

class ConfirmFinalOrderActivity : AppCompatActivity() {
    private lateinit var nameEditText: EditText
    private var phoneEditText: EditText? = null
    private var addressEditText: EditText? = null
    private var cityEditText: EditText? = null
    private lateinit var confirmOrderBtn: Button
    private  var totalAmount: String? = ""


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_confirm_final_order)

        totalAmount = intent.getStringExtra("Total price")

        Toast.makeText(this, "Total Ammount :  $totalAmount$", Toast.LENGTH_LONG).show()
        confirmOrderBtn = findViewById(R.id.confirm_final_order_btn)
        nameEditText = findViewById(R.id.shippment_name)
        phoneEditText = findViewById(R.id.shippment_phone_number)
        addressEditText = findViewById(R.id.shippment_address)
        cityEditText = findViewById(R.id.shipment_city)
        val databaseReference = FirebaseDatabase.getInstance().reference.child("Users").child(
            Prevalent.currentOnlineUser!!.phone!!
        )
        databaseReference.addValueEventListener(object : ValueEventListener {
            override fun onDataChange(dataSnapshot: DataSnapshot) {
                val name = dataSnapshot.getValue(Users::class.java)!!.name
                nameEditText.setText(name)
            }

            override fun onCancelled(databaseError: DatabaseError) {
            }
        })

        confirmOrderBtn.setOnClickListener(View.OnClickListener { check() })
    }


    private fun check() {
        if (TextUtils.isEmpty(nameEditText!!.text.toString())) {
            nameEditText!!.error = "Please provide your full name"
        } else if (TextUtils.isEmpty(phoneEditText!!.text.toString())) {
            phoneEditText!!.error = "Please provide your phone number"
        } else if (TextUtils.isEmpty(addressEditText!!.text.toString())) {
            addressEditText!!.error = "Please provide your phone number"
        } else if (TextUtils.isEmpty(cityEditText!!.text.toString())) {
            cityEditText!!.error = "Please provide your phone number"
        } else {
            confirmOrder()
        }
    }

    private fun confirmOrder() {
        val saveCurrentDate: String
        val saveCurrentTime: String
        val callForDate = Calendar.getInstance()
        val currentDate = SimpleDateFormat("MMM dd, yyyy")
        saveCurrentDate = currentDate.format(callForDate.time)
        val currentTime = SimpleDateFormat("HH:mm:ss a")
        saveCurrentTime = currentDate.format(callForDate.time)

        val ordersRef = FirebaseDatabase.getInstance().reference
            .child("Orders")
            .child(Prevalent.currentOnlineUser!!.phone!!)
        val ordersMap = HashMap<String, Any?>()
        ordersMap["totalAmount"] = totalAmount
        ordersMap["name"] = nameEditText!!.text.toString()
        ordersMap["phone"] = phoneEditText!!.text.toString().toString()
        ordersMap["adress"] = addressEditText!!.text.toString()
        ordersMap["city"] = cityEditText!!.text.toString()
        ordersMap["date"] = saveCurrentDate
        ordersMap["time"] = saveCurrentTime
        ordersMap["state"] = "not shipped"


        ordersRef.updateChildren(ordersMap).addOnCompleteListener { task ->
            if (task.isSuccessful) {
                FirebaseDatabase.getInstance().reference
                    .child("Cart List")
                    .child("User View")
                    .child(Prevalent.currentOnlineUser!!.phone!!)
                    .removeValue()
                    .addOnCompleteListener { task ->
                        if (task.isSuccessful) {
                            Toast.makeText(
                                this@ConfirmFinalOrderActivity,
                                "Your final Order has been placed successfully",
                                Toast.LENGTH_LONG
                            ).show()
                            val intent =
                                Intent(this@ConfirmFinalOrderActivity, HomeActivity::class.java)
                            intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK)
                            finish()
                        }
                    }
            }
        }
    }
}
